signal-hook-tokio/src/lib.rs

```diff
@@ -73,179 +114,37 @@ macro_rules! implement_signals_with_pipe {
-                PollResult::Signal(result) => Poll::Ok(Async::Ready(Some(result))),
-                PollResult::Closed => Poll::Ok(Async::Ready(None)),
-                PollResult::Err(error) => Poll::Err(error),
-            }
+impl Signals {
+    fn has_signals(read: &mut UnixStream, ctx: &mut Context<'_>) -> Result<bool, Error> {
+        let mut buf = [0u8];
+        let mut read_buf = ReadBuf::new(&mut buf);
+        match Pin::new(read).poll_read(ctx, &mut read_buf) {
+            Poll::Pending => Ok(false),
+            Poll::Ready(Ok(())) => Ok(true),
+            Poll::Ready(Err(error)) => Err(error),
         }
     }
 }
 
-/// A module for integrating signal handling with the Tokio 0.3 runtime.
-///
-/// This provides the [`Signals`][v0_3::Signals] struct which acts as a
-/// [`Stream`][`futures_0_3::stream::Stream`] of signals.
-///
-/// # Example
-///
-/// ```rust
-/// # use tokio_0_3 as tokio;
-/// # use futures_0_3 as futures;
-///
-/// use std::io::Error;
-///
-/// use signal_hook::consts::signal::*;
-/// use signal_hook_tokio::v0_3::Signals;
-///
-/// use futures::stream::StreamExt;
-///
-/// async fn handle_signals(signals: Signals) {
-///     let mut signals = signals.fuse();
-///     while let Some(signal) = signals.next().await {
-///         match signal {
-///             SIGHUP => {
-///                 // Reload configuration
-///                 // Reopen the log file
-///             }
-///             SIGTERM | SIGINT | SIGQUIT => {
-///                 // Shutdown the system;
-///             },
-///             _ => unreachable!(),
-///         }
-///     }
-/// }
-///
-/// #[tokio::main]
-/// async fn main() -> Result<(), Error> {
-///     let signals = Signals::new(&[
-///         SIGHUP,
-///         SIGTERM,
-///         SIGINT,
-///         SIGQUIT,
-///     ])?;
-///     let handle = signals.handle();
-///
-///     let signals_task = tokio::spawn(handle_signals(signals));
-///
-///     // Execute your main program logic
-///
-///     // Terminate the signal stream.
-///     handle.close();
-///     signals_task.await?;
-///
-///     Ok(())
-/// }
-/// ```
-#[cfg(feature = "support-v0_3")]
-pub mod v0_3 {
-    use std::pin::Pin;
-
-    use futures_0_3::stream::Stream;
-    use futures_0_3::task::{Context, Poll};
-    use signal_hook::iterator::backend::PollResult;
-    use tokio_0_3::io::{AsyncRead, ReadBuf};
-    use tokio_0_3::net::UnixStream;
-
-    implement_signals_with_pipe!(UnixStream);
-
-    impl Signals {
-        fn has_signals(read: &mut UnixStream, ctx: &mut Context<'_>) -> Result<bool, Error> {
-            let mut buf = [0u8];
-            let mut read_buf = ReadBuf::new(&mut buf);
-            match Pin::new(read).poll_read(ctx, &mut read_buf) {
-                Poll::Pending => Ok(false),
-                Poll::Ready(Ok(())) => Ok(true),
-                Poll::Ready(Err(error)) => Err(error),
-            }
-        }
-    }
+impl Stream for Signals {
+    type Item = c_int;
```

Comment:
Seems I've missed this part when adding the support for extended info :-|. Well, fixing that doesn't belong into this PR, so let's ignore it for now.